lesson18/main.js

```diff
@@ -132,22 +144,81 @@ const switchDisableForBtn = () => {
     backBtn.disabled = false;
   }
 
-  if (getCurrentImgIndex() === imgArray.length - 1) {
+  if (getCurrentImgIndex() === images.length - 1) {
     nextBtn.disabled = true;
   } else {
     nextBtn.disabled = false;
   }
 };
 
-const createPagination = () => {
+const createNumPagination = () => {
   const pagination = createElementWithClassName("p", "number-pagination");
   slideshowWrap.insertAdjacentElement("afterend", pagination);
   getCurrentPageNum();
 };
 
 const getCurrentPageNum = () => {
   const pagination = document.querySelector(".number-pagination");
-  pagination.textContent = `${getCurrentImgIndex() + 1}/${imgArray.length}`;
+  pagination.textContent = `${getCurrentImgIndex() + 1}/${images.length}`;
+};
+
+const createIndicatorForSlideshow = () => {
+  createIndicator();
+  addEventToIndicator();
+};
+
+const createIndicator = () => {
+  const pagination = createElementWithClassName("div", "indicators");
+  const fragment = document.createDocumentFragment();
+  for (let i = 0; i < images.length; i++) {
+    const indicator = createElementWithClassName("sapn", "indicator");
+    //最初のインデックスをアクティブにする
+    i === 0 && indicator.classList.add("is-active");
+    fragment.appendChild(indicator);
+  }
+  pagination.appendChild(fragment);
+  slideshowWrap.insertAdjacentElement("afterend", pagination);
+};
+
+const addEventToIndicator = () => {
+  const indicator = document.querySelector(".indicators");
+  indicator.addEventListener(
+    "click",
+    (e) => {
+      clickIndicatorEvents(e);
+    },
```

Comment:
```suggestion
      clickIndicatorEvents;,
```

Reply:
I rewrote it to one line. If I remove `e`, indicators don't work by themselves so  I left `e` here.
Thank you :)
[02dad55](https://github.com/yuka830/js-lessons/pull/23/commits/02dad555bfef3235e57d49cae605f23cb1f86111)